### ISSUE TITLE
Missing buildings using GeoClimate Tools

### ISSUE THREAD

**Issue by sachabaclet** [orbisgis/geoclimate#997]
Hi,

I have used `GeoClimate_Tools-1.0.1` to extract the buildings in the city of Tartu (Estonia). However, I noticed that some buildings were missing from the output `building.geojson` file.

You can see in the following image the buildings extracted by the tools in pink and the missing buildings in blue.

![image](https://github.com/user-attachments/assets/692a350e-f124-4229-91ef-659600ff234d)

**Comment by ebocher** [comment]
Hi @sachabaclet 

Please share the json file you are using with GeoClimate so we can debugg it.

Thanks

**Comment by sachabaclet** [comment]
Here is the geojson file generated by GeoClimate Tools:
[building.zip](https://github.com/user-attachments/files/17128009/building.zip)

**Comment by sachabaclet** [comment]
I am using the tool by Samuel Marsault, so this may be what you are looking for:
https://github.com/Universite-Gustave-Eiffel/plamade/blob/osm/src/main/groovy/GeoClimate_Script_CL.groovy

**Comment by wurtzj** [comment]
Hello @j3r3m1,

I think tents should not be added for the moment has it has no thermal inertia, even if it is impervious (which will be considered as road in meteorological models).

**Comment by pierromond** [comment]
Hello @wurtzj ,

For the acoustic use of geoclimate is important to use every buildings... Maybe a option could be added ?

Best

**Comment by wurtzj** [comment]
Maybe it could be added in an additional class ? TENT_FRACTION ?

**Comment by ebocher** [comment]
Good job inspecteur @j3r3m1 !
GeoClimate's model must satisfy as many uses as possible.
Let's take our time before fixing this issue ;-)

**Comment by wurtzj** [comment]
Hello @j3r3m1 and @ebocher ,

Does Geoclimate take into account building=warehouse ?
It seems to me it is lacking.

Cheers,

Jean

**Comment by j3r3m1** [comment]
Do you have test cases where you observed or suppose it might be missing ? Because it should normally work if the building has the key:value "building:warehouse" (cf. https://github.com/orbisgis/geoclimate/blob/master/osm/src/main/resources/org/orbisgis/geoclimate/osm/buildingParams.json)

**Comment by wurtzj** [comment]
![Capture d’écran du 2024-10-03 13-42-56](https://github.com/user-attachments/assets/9d71c0d3-3992-4553-b4eb-a2cacce8c809)
Here is a harbour near Seine river. Building fraction is 14% but looking at Open Street Map it seems higher to me. But Maybe I'm wrong ? 
![Capture d’écran du 2024-10-03 13-44-10](https://github.com/user-attachments/assets/f86844cc-87f9-4c32-8756-3558459f1094)

**Comment by wurtzj** [comment]
Yes to me building fraction is underestimated.
But building fraction do include industrial building ?

**Comment by j3r3m1** [comment]
Yes of course industrial buildings are considered in the building fraction. Here is the data at RSU scale. I do not think the building fraction is wrong, even though there is a quite high spatial variability.

![image](https://github.com/user-attachments/assets/f9d79a7a-f9ab-412b-9750-8e63e3d162b1)![image](https://github.com/user-attachments/assets/aeef3856-c3e8-453a-b17d-fcd310af1d2a)
![image](https://github.com/user-attachments/assets/7abd0ca8-efa6-4006-bdc4-120604f99549)

**Comment by wurtzj** [comment]
Ok good to know. I think it is a cognitive bias from my side then.
Thanks !

**Comment by j3r3m1** [comment]
I am going to test if we get all buildings from the zone anyway, we might have forgotten some tags (I know understand that it was your initial question...), do not close it yet.

**Comment by j3r3m1** [comment]
We might think that there are several missing buildings in the area (cf. below)
![image](https://github.com/user-attachments/assets/1716d54b-1187-48af-8fba-5bba0eaf1616)
![image](https://github.com/user-attachments/assets/db05619b-f605-4371-b9b5-8ff5077708e8)

But actually most of the polygons that look like buildings are construction zones or brownfields and their color is slightly different than the buidings. There is a single building that is missing and need to be added but it is tiny (https://www.openstreetmap.org/way/796915590).
Its tag is "building":"storage_tank".

**Comment by wurtzj** [comment]
Thanks for checking ! :)

**Comment by j3r3m1** [comment]
As a summary of this issue, we should add:
- an urban area: "landuse:garage"
- many types of buildings: "building:garage",  "building:civic",  "building:service",  "building:manufacture",  "building:storage_tank"

Each building should have a corresponding type for the area fraction calculation: https://geoclimate.readthedocs.io/en/latest/RSU-indicators.html#area-fraction-x. I propose:
- garage --> "non_heated"
- civic --> "tertiary"
- service --> "non_heated"
- manufacture --> "light_industrial"
- storage_tank --> "non_heated"

The documentation should be modified accordingly (https://geoclimate.readthedocs.io/en/latest/Building-layer.html#type-and-main-use-column-values AND https://geoclimate.readthedocs.io/en/latest/RSU-indicators.html#area-fraction-x)
